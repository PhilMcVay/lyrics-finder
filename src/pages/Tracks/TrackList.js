import React, { useState } from "react";

import Track from "./Track";
import SearchFilter from "./SearchFilter";
import { TrackListContainer } from "./TrackList.styles";

const TrackList = ({ trackList = [], artistName }) => {
  const [searchFilter, setSearchFilter] = useState("");

  const handeSearchFilter = ({ target: { value } }) => setSearchFilter(value);

  return (
    <>
      {trackList.length > 0 && (
        <SearchFilter
          placeholder="Filter Tracks..."
          value={searchFilter}
          onChange={handeSearchFilter}
        />
      )}
      <TrackListContainer>
        {trackList
          .filter((track) =>
            track.toLowerCase().includes(searchFilter.toLowerCase().trim())
          )
          .sort()
          .map((track) => (
            <Track track={track} artistName={artistName} />
          ))}
      </TrackListContainer>
    </>
  );
};

export default TrackList;
