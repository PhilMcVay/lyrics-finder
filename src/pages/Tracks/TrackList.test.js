import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import TrackList from "./TrackList";

describe("<TrackList>", () => {
  it("Should Render", () => {
    const { container } = render(<TrackList />);

    expect(container).toBeInTheDocument();
  });

  it("Should show a list of tracks", () => {
    render(
      <BrowserRouter>
        <TrackList trackList={["a-track", "b-track", "c-track"]} />
      </BrowserRouter>
    );

    expect(screen.getByText("a-track")).toBeInTheDocument();
    expect(screen.getByText("b-track")).toBeInTheDocument();
    expect(screen.getByText("c-track")).toBeInTheDocument();
  });

  it("Should not display the filter search if no tracks are provided", () => {
    render(
      <BrowserRouter>
        <TrackList trackList={[]} />
      </BrowserRouter>
    );

    expect(screen.getByTestId("tracks-list")).toMatchSnapshot();
  });

  it("Should filter the track list based on the search filter", () => {
    render(
      <BrowserRouter>
        <TrackList trackList={["a-track", "b-track", "c-track"]} />
      </BrowserRouter>
    );

    expect(screen.getByText("a-track")).toBeInTheDocument();
    expect(screen.getByText("b-track")).toBeInTheDocument();
    expect(screen.getByText("c-track")).toBeInTheDocument();

    userEvent.type(screen.getByPlaceholderText("Filter Tracks..."), "b");

    expect(screen.queryByText("a-track")).not.toBeInTheDocument();
    expect(screen.getByText("b-track")).toBeInTheDocument();
    expect(screen.queryByText("c-track")).not.toBeInTheDocument();
  });
});
