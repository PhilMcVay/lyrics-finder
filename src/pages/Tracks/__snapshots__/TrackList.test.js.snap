// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackList> Should not display the filter search if no tracks are provided 1`] = `
<ul
  class="sc-dlnjPT dWicHX"
  data-testid="tracks-list"
/>
`;
